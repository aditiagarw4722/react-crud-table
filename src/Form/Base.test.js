import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import FormBase from './Base';

jest.mock('formik', () => ({
  Field: 'Field',
}));

import { Field } from 'formik';

const renderer = new ShallowRenderer();

describe('Form', () => {
  let props;

  beforeEach(() => {
    props = {
      data: {
        title: 'Test title',
        fields: [
          {
            name: 'name',
            label: 'Label',
            placeholder: 'Write name',
            render: () => 'Name',
            type: 'text',
            readOnly: false,
          },
        ],
        message: 'Hello world',
        generalErrorMessage: 'General error',
        submitButtonProps: { className: 'Cool' },
        submitText: 'Submit',
      },
      onSubmit: jest.fn(() => Promise.resolve({ success: true })),
      initialValues: { name: 'John' },
      touched: { name: true },
      errors: {
        name: 'Invalid name',
      },
      error: 'There is an error',
    };
  });

  it('renders as expected with the given props', () => {
    const result = renderer.render(<FormBase {...props} />);
    expect(result).toMatchSnapshot();
  });
});
