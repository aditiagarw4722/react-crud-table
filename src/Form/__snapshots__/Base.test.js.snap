// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders as expected with the given props 1`] = `
<FormikForm>
  <Message>
    Hello world
  </Message>
  <ErrorMessage>
    There is an error
  </ErrorMessage>
  <ErrorMessage>
    General error
  </ErrorMessage>
  <FieldContainer>
    <Label
      htmlFor="name"
    >
      Label
    </Label>
    <Field
      name="name"
      placeholder="Write name"
      readOnly={false}
      render={[Function]}
      type="text"
    />
    <FieldError>
      Invalid name
    </FieldError>
  </FieldContainer>
  <ErrorMessage>
    General error
  </ErrorMessage>
  <Button
    className="Cool"
    modifiers="positive"
    type="submit"
  >
    Submit
  </Button>
</FormikForm>
`;
